projects/rocr-runtime/runtime/hsa-runtime/core/inc/runtime.h

```diff
@@ -595,6 +596,63 @@ class Runtime {
     std::vector<void*> arg_;
   };
 
+  // Event item structure to hold all signal information
```

Comment:
I think we can remove these elements from "struct AsyncEvents":

std::vector<hsa_signal_t> signal_;
std::vector<hsa_signal_condition_t> cond_;
std::vector<hsa_signal_value_t> value_;
std::vector<hsa_amd_signal_handler> handler_;
std::vector<HsaEvent*> hsa_events_; //!< A list of HSA events for KFD wait
std::vector<uint64_t> age_;         //!< The age list for KFD wait


And only use the new list in ConcurrentAsyncEvents->event_queue_ to track the list of events. This will behave like a AoS, which is better for caching.